### ISSUE TITLE
Laravel 9 Casts/Attribute handling

### ISSUE THREAD

**Issue by vanyamil** [nunomaduro/larastan#1153]
- Larastan Version: 2.1.1
- `--level` used: 5
- Pull request with failing test: Not tech-savvy enough for that

### Description

The new approach in Laravel 9 to handle model properties' accessors and/or mutators, using the class `Illuminate\Database\Eloquent\Casts\Attribute`, needs proper handling. Right now, all such properties are interpreted as strings.

### Laravel code where the issue was found

Create a model with a string/varchar column where you store space-separated list of strings (example: scopes of an OAuth token). Then, add the Attribute cast to the model, something like this:

```php
	/**
	 * The list of scopes is stored as a space-separated string.
	 */
	public function scopes() : Attribute
	{
		return Attribute::make(
			get: fn(?string $value) => is_null($value) ? [] : explode(' ', $value),
			set: function(array $value) {
				$set = array_unique($value);
				sort($set);
				return ['scopes' => implode(' ', $set)];
			}
		);
	}
```

Then, trying to use the accessor value in something like `in_array('x', $model->scopes)` raises the error 

```
Parameter #2 $haystack of function in_array expects array, string given. 
```

And trying to set the value to an array, for example `$model->scopes = ['x']`, raises the error

```
Property App\Models\CustomModel::$scopes (string) does not accept array<int, string>.
```

**Comment by vanyamil** [comment]
At the moment, this can be solved by manually supplying a `@property` PHPDoc comment on the model, with the intended type (received from accessor - sent to mutator) set.

**Comment by vanyamil** [comment]
Furthermore, an accessor not based on an existing attribute of the model (such as $model->fullName for a User model that has first_name and last_name columns) is not detected as an allowed property on the model.

**Comment by canvural** [comment]
Hi,

Yes, new Laravel 9 style attributes are not supported yet. So this is a feature request.

**Comment by canvural** [comment]
This is implemented in #1156 and released in https://github.com/nunomaduro/larastan/releases/tag/v2.1.2

To be able to use this correctly now you need to add return type to your accessors like so
```php
/** @return Attribute<string[], string[]> */
public function scopes() : Attribute
{
	return Attribute::make(
		get: fn(?string $value) => is_null($value) ? [] : explode(' ', $value),
		set: function(array $value) {
			$set = array_unique($value);
			sort($set);
			return ['scopes' => implode(' ', $set)];
		}
	);
}
```

(I put `string[]` as an example, you can correct it. First generic type is for the `get` and second is for `set`)
